lib/index.js

```diff
@@ -149,28 +160,37 @@ export default {
           linterName => configEnabledEngines.has(linterName));
 
         // Construct the command line invocation which runs the Code Climate CLI
-        const cmd = [this.executablePath, 'analyze',
-          '-f json',
-          makeEngineString(linterEnabledEngines),
-          `'${atom.project.relativize(filePath)}'`,
-          '< /dev/null'].join(' ');
+        const execArgs = ['analyze', '-f', 'json'].concat(
+          getEngineArgs(linterEnabledEngines),
+          [atom.project.relativize(filePath), '</dev/null']);
+        const execOpts = {
+          cwd: dirname(configurationFilePath),
+        };
+        if (this.disableTimeout) {
+          execOpts.timeout = Infinity;
+        }
 
         // Debug the command executed to run the Code Climate CLI to the console
         if (atom.inDevMode()) {
           // eslint-disable-next-line no-console
-          console.log(`linter-codeclimate:: Command: \`${cmd}\``);
+          console.log('linter-codeclimate:: Command: ' +
+            `\`${this.executablePath} ${execArgs.join(' ')}\``);
         }
 
         // Start measure for how long the analysis took.
         startMeasure('linter-codeclimate: Analysis');
 
         // Execute the Code Climate CLI, parse the results, and emit them to the
         // Linter package as warnings. The Linter package handles the styling.
-        const execArgs = ['-lc', cmd];
-        const execOpts = {
-          cwd: dirname(configurationFilePath),
-        };
-        const result = await Helpers.exec('/bin/bash', execArgs, execOpts);
+        let result;
+        try {
+          result = await Helpers.exec(this.executablePath, execArgs, execOpts);
+        } catch (e) {
+          if (ccBinaryFound(this.executablePath, e)) {
+            throw e;
+          }
+          return [];
```

Comment:
Return `null` here, this may be a temporary issue so we should let `linter` keep managing the state of any messages that were currently there. If the code never had messages this doesn't change anything 😉.